pkg/api/v20210131preview/openshiftcluster_validatestatic.go

```diff
@@ -262,6 +273,18 @@ func (sv *openShiftClusterStaticValidator) validateWorkerProfile(path string, wp
 	if wp.Count < 2 || wp.Count > 50 {
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".count", "The provided worker count '%d' is invalid.", wp.Count)
 	}
+	if wp.DiskEncryptionSetID != "" {
+		if !validate.RxDiskEncryptionSetID.MatchString(wp.DiskEncryptionSetID) {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".diskEncryptionSetId", "The provided worker disk encryption set '%s' is invalid.", wp.DiskEncryptionSetID)
+		}
+		desr, err := azure.ParseResourceID(wp.DiskEncryptionSetID)
+		if err != nil {
+			return err
+		}
+		if desr.SubscriptionID != sv.r.SubscriptionID {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".diskEncryptionSetId", "The provided worker disk encryption set '%s' is invalid: must be in same subscription as cluster.", wp.DiskEncryptionSetID)
+		}
+	}
```

Comment:
Note that currently we allow two different disk encrypthion sets to be used: one for master and one for worker nodes.

We can change the static validation to make sure that master and worker profiles use the same disk encryption set. I'm however not sure whether it makes sense to impose this restriction or not. Opinions?

Reply:
I can't think of a reason why it would need to be the same.

Reply:
> I can't think of a reason why it would need to be the same.

Did you mean different? I would imagine that most of the customers will be using the same disk encryption set for both master and worker nodes. But there are probably use cases for having different sets for workers. Maybe for multi tenant clusters?

Reply:
Keep them separate. If there is a possibility - somebody will take it.

Reply:
We had a discussion with MJ about it and decided to force master and worker disk encryption sets to be the same for now in static validation.

We can relax this validation later, if we see a use case.